### ISSUE TITLE
Field description on output objects

### ISSUE THREAD

**Issue by piotr-swierczynski** [leangen/graphql-spqr#137]
Hi! I've got a question with regards to field description. I'm able to put description on Input object with @GraphQLInputField. So if I have object called Dog, I can put descriptions on DogInput fields, but not on Dog fields. Is there any simple solution to put description on output object field (annotation) or only workaround is too use Custom Output converter?

**Comment by kaqqao** [comment]
You use the regular `@GraphQLQuery` for that. In GraphQL, all queries are really just fields, so there's no distinction. Even the root queries are just fields on the query root object. That's why there's no specal annotation for that.

**Comment by piotr-swierczynski** [comment]
Ok, that works with no doubt. But when i put @GraphQLInputField on getter it does not change a field name. For instance, getId remains as id in ObjectInput, but in Object with @GraphQLQuery it transforms into getId. Is there an easy way to keep just field name without explicitly putting it in GraphQLQuery annotation argument? (Can put annotation over field, because its inherited class private field)

**Comment by piotr-swierczynski** [comment]
Ok, I've solved my problem with a custom type mapper. Not considering my previous issue, I wanted to use messages from messages.properties (Spring Boot App) for descriptions and couldn't figured it out how to change ${my.message.key} in @GraphQLQuery description to message from the file in more elegant way then using designed TypeMapper.

**Comment by kaqqao** [comment]
Hey, sorry for not replying.
The idea here is to use a custom `OperationNameGenerator` which you can then set on the `ResolverBuilder`s that you're using. That's simpler and less invasive than a custom mapper, if all you need to change the naming policy. The implementation that simply returns the method name is [`MethodOperationNameGenerator`](https://github.com/leangen/graphql-spqr/blob/582d405029eb57b21623785d90b17dd1ae78a572/src/main/java/io/leangen/graphql/metadata/strategy/query/MethodOperationNameGenerator.java). Maybe you could make your own that does something like:

```java
@Override
    public String generateQueryName(Method queryMethod, AnnotatedType type, Object instance) {
        return ClassUtils.isGetter(queryMethod) ? ClassUtils.getFieldNameFromGetter(queryMethod) : queryMethod.getName();
    }
```

Depending on your `OperationNameGenerator` implementation, it could also interpolate the names based on a messages file etc.

Now that you mention it, I think this would be a very good feature out of the box in the SPQR Spring Starter project 😃 

I'll also make a small tweak to SPQR itself to drop the `get` from method names in nested objects.

**Comment by kaqqao** [comment]
@piotr-swierczynski Just an update. In issue #146, I added support for string interpolation (to be released with 0.9.8).

To use it, call:
```java
GraphQLSchemaGenerator#withStringInterpolation(MessageBundle...)
```

To keep SPQR framework-agnostic, the feature works via a custom (but minimal) `MessageBundle` interface. Two simple implementation are provided, but you'd likely want to provide an implementation that makes the most sense in your environment. E.g. for Spring, you'd want it to be backed by Spring's `MessageSource`.

I'll add an out-of-the-box integration with Spring in the [SPQR Spring Boot Starter project](https://github.com/leangen/graphql-spqr/issues/137).

**Comment by kaqqao** [comment]
I've also refactored the provided `OperationNameGenerator`s significantly, so that any `OperationNameGenerator` can be freely mixed with any `ResolverBuilder`.


Also, the new `DefaultOperationNameGenerator` now has the behavior you're after, so no need for any customization in v0.9.8.
